ui/app/pages/unlock-page/unlock-page.component.js

```diff
@@ -39,12 +41,36 @@ export default class UnlockPage extends Component {
     }
   }
 
+  componentDidMount () {
+    const { isNotification } = this.props
+    if (isNotification) {
+      window.addEventListener('beforeunload', this.beforeUnload)
+    }
+  }
+
+  beforeUnload = () => {
+    const {
+      isUnlocked,
+      rejectAllPermissionsRequests,
+    } = this.props
+
+    // We only want to reject requests in the event that
+    // a user closes the popup without unlocking.
+    if (!isUnlocked) {
+      rejectAllPermissionsRequests()
+    }
+  }
+
+  removeBeforeUnload = () => {
```

Comment:
does `removeBeforeUnload` need to be bound to `this` the same way `beforeUnload` does?

Reply:
@marshall - just to make sure I understand your feedback, are you saying just to call `window.removeEventListener('beforeunload', this.beforeUnload)` directly rather than having a class Method `removeBeforeUnload`, or?

Reply:
apologies, I think my feedback here was based on a bad assumption about arrow function behavior.. namely that they are bound to `this` during declaration, which after checking up on MDN is apparently not correct.

this comment is safe to ignore :)